LiteDB/Engine/Services/TransactionService.cs

```diff
@@ -75,17 +75,32 @@ public void PersistDirtyPages()
 
                 // mark header as recovery before start writing (in journal, must keep recovery = false)
                 header.Recovery = true;
+
+                // flush to disk to ensure journal is committed to disk before proceeding
+                _disk.Flush();
```

Comment:
`WriteJournal` method contains a a `Flush` call at end:

https://github.com/mbdavid/LiteDB/blob/v4/LiteDB/Engine/Disks/FileDiskService.cs#L194

Reply:
Yes, but `header.Recovery = true` is lost if the write is interrupted after that line. As there is no guarantee to the order of writes, if the changes below are committed but this is not, then you end up with a corrupt database.

Reply:
You right, I miss this. I will accept your PR, thanks. New 4.1.5 will be released soon.

Reply:
Hi, if I already have a corrupted database from power failure, how can I fix it?